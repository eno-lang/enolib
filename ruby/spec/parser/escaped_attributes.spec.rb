# frozen_string_literal: true

input = <<~DOC.strip
field:
`key`=value
``k`ey`` = value
```ke``y```     = more value
    `` `key` ``     =
DOC

describe Enolib::Parser do
  describe 'Escaped attributes' do
    context = Enolib::Context.new(input)

    it 'parses as expected' do
      expect(context.document).to match_snapshot
    end
  end
end
