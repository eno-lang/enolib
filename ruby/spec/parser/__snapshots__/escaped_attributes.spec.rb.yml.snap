---
"./spec/parser/escaped_attributes.spec.rb[1:1:1]": &1
  :elements:
  - &2
    :index: 0
    :line: 0
    :ranges:
      :line:
      - 0
      - 6
      :field_operator:
      - 5
      - 6
      :key:
      - 0
      - 5
    :key: field
    :type: :field
    :parent: *1
    :attributes:
    - :index: 7
      :line: 1
      :ranges:
        :line:
        - 7
        - 18
        :attribute_operator:
        - 12
        - 13
        :escape_begin_operator:
        - 7
        - 8
        :escape_end_operator:
        - 11
        - 12
        :key:
        - 8
        - 11
        :value:
        - 13
        - 18
      :type: :attribute
      :key: key
      :value: value
      :parent: *2
    - :index: 19
      :line: 2
      :ranges:
        :line:
        - 19
        - 35
        :attribute_operator:
        - 28
        - 29
        :escape_begin_operator:
        - 19
        - 21
        :escape_end_operator:
        - 25
        - 27
        :key:
        - 21
        - 25
        :value:
        - 30
        - 35
      :type: :attribute
      :key: k`ey
      :value: value
      :parent: *2
    - :index: 36
      :line: 3
      :ranges:
        :line:
        - 36
        - 64
        :attribute_operator:
        - 52
        - 53
        :escape_begin_operator:
        - 36
        - 39
        :escape_end_operator:
        - 44
        - 47
        :key:
        - 39
        - 44
        :value:
        - 54
        - 64
      :type: :attribute
      :key: ke``y
      :value: more value
      :parent: *2
    - :index: 65
      :line: 4
      :ranges:
        :line:
        - 65
        - 86
        :attribute_operator:
        - 85
        - 86
        :escape_begin_operator:
        - 69
        - 71
        :escape_end_operator:
        - 78
        - 80
        :key:
        - 72
        - 77
      :type: :attribute
      :key: "`key`"
      :parent: *2
  :type: :document
